lib/stream.js

```diff
@@ -51,15 +51,32 @@ Stream.isReadable = utils.isReadable;
 Stream.Readable = require('internal/streams/readable');
 for (const key of ObjectKeys(streamReturningOperators)) {
   const op = streamReturningOperators[key];
-  Stream.Readable.prototype[key] = function(...args) {
-    return Stream.Readable.from(ReflectApply(op, this, args));
-  };
+  // name function
+  const fn = {
+    [op.name]: function(...args) {
```

Comment:
```suggestion
    [op.name](...args) {
```
nit

Reply:
Note that this difference i believe is observable with the .prototype property on the resulting function.